charts/consul/templates/connect-inject-deployment.yaml

```diff
@@ -125,6 +127,19 @@ spec:
                 -release-name="{{ .Release.Name }}" \
                 -release-namespace="{{ .Release.Namespace }}" \
                 -listen=:8080 \
+                {{- if and .Values.externalServers.enabled (not .Values.externalServers.hosts) }}{{ fail "externalServers.hosts must be set if externalServers.enabled is true" }}{{ end -}}
+                {{- if .Values.externalServers.enabled }}
+                {{- if .Values.global.tls.enabled }}
+                -use-https \
+                {{- end }}
+                {{- range .Values.externalServers.hosts }}
+                -server-address={{ quote . }} \
+                {{- end }}
+                -server-port={{ .Values.externalServers.httpsPort }} \
```

Comment:
Yeah good question! I'll need to consolidate this logic in the future to make it more consistent and also so we don't have to keep adding flags to each command. I'll add it as a todo in the future but for now will keep it as is because if you set this flag when TLS is disabled, it had no effect, so user should see no difference.